django/db/models/query.py

```diff
@@ -452,8 +453,10 @@ def bulk_create(self, objs, batch_size=None):
                     ids = self._batched_insert(objs_without_pk, fields, batch_size)
                     if connection.features.can_return_ids_from_bulk_insert:
                         assert len(ids) == len(objs_without_pk)
-                    for i in range(len(ids)):
-                        objs_without_pk[i].pk = ids[i]
+                    for obj_without_pk, pk in zip(objs_without_pk, ids):
```

Comment:
I suggested using zip now that we do multiple index lookips in `objs_without_pk` which we can assume to be relatively large given `bulk_create` was used in favor of `create`.